projects/frontend/data-pipelines/gui/projects/ui/src/app/app.component.ts

```diff
@@ -54,6 +54,10 @@ export class AppComponent implements OnInit {
         return this.oauthService.getIdentityClaims() ? this.getIdentityClaim('username') : 'N/A';
     }
 
+    get explorePageVisible(): boolean {
+        return !this.appConfigService.getConfig().ignoreComponents.includes('explorePage');
```

Comment:
seems you are now verifying the value of appConfig.json propagates to the component, when you should rather verify the value with the functional expectation

Reply:
I don't understand what you're think should be changed here.

Reply:
nit: We now have a proper toggle able visibility testing in e2e tests, right? 
That's very cool.

I'll try to explain, 
in `projects/frontend/data-pipelines/gui/projects/ui/src/app/app.component.spec.ts`
on L101 and L102 you have disabled ignore, then
on L112 you evaluate the code here L67 that reads the config and returns the value.


So, the test in L112 named 'should create the app with no components ignored'
does actually perform a test if the config value propagated empty? Why? It tests the config property, rather than actually testing if page is visible, is what I'm saying.


The test looks like:
```
explorePageVisibleConfig = empty
expect(explorePageVisibleConfig) is empty
```
do we need it?

Reply:
I don't think we can test if the html renders using an Angular spec. Maybe I'll move this scenario to cypress as well.

Reply:
It's possible to test html rendering in unit tests in Angular.
You could look into `placeholder.service.spec.ts` for examples, but in short we access `debugElement` from fixture or some other queried element and using `query` method we could select something from the html nodes.
e.g.

```
const gridDebugElement: DebugElement = fixture.debugElement.query(By.directive(ClarityDataGridComponentStub));
gridComponent = gridDebugElement.componentInstance;
```

Predicate provided to `query` method could be something of `By.directive` where we provide class reference of Component of Directive or `By.css` where we provide name of the html node (e.g. `div`, `p`, `span`, `lib-data-jobs-explore-grid`, etc...)

Reply:
I tried doing it using cypress, but it increased the runtime beyond the 2 hour limit. Maybe what @gorankokin suggested is the way to go. Adding it as tech debt, because I've already spent too much time on it. 

https://jira.eng.vmware.com/browse/TAUR-6037